experimental/bits/simd.h

```diff
@@ -4966,6 +4967,15 @@ template <typename _Tp, typename _Abi>
 	  _Impl::_S_load(_Flags::template _S_apply<simd>(__mem), _S_type_tag))
       {}
 
+    // gather constructor
+    template <typename _Up, typename _Flags>
+      _GLIBCXX_SIMD_ALWAYS_INLINE
+      simd(const _Up* __mem, const __int_for_sizeof_t<_Up>* __idx, _Flags)
```

Comment:
* This is a non-standard (as not specified in an ISO standard) interface. I believe we therefore should implement gather & scatter interfaces only as free functions (possibly hidden friends) or static member functions. And the function name must be a reserved name.
* I think we want the index parameter to always be of type `simd`. Turning an array into simd isn't too hard. And the expectation for gather & scatter is that indexes are computed via `simd` objects.
* Do you know whether alignment of the memory pointer matters for any gather / scatter hardware? Because if not then the `_Flags` parameter is unnecessary. I'd drop it.
* A `constexpr` scale parameter has been useful in Vc 1. I'm not sure whether we want it as public API at some point, though.

```c++
template <size_t _Scale = 1, typename _From, typename _Idx, typename _IdxAbi>
  _GLIBCXX_SIMD_ALWAYS_INLINE static _GLIBCXX_SIMD_CONSTEXPR
  simd
  __gather(const _From* __mem, const simd<_Idx, _IdxAbi>& __idx, _SizeConstant<_Scale> __scale = {})
  {
    return simd([&](auto __i) { return static_cast<value_type>(__mem[__idx[__i] * __scale]); });
  }
```

I also just implemented a complete `_S_gather` implementation for `_SimdImplX86` (AVX512 and AVX2) and fallback in `_SimdImplBuiltin`. I didn't implement anything else, because I only use it internally to implement a lookup table for my `std::exp(simd)` implementation. It's not pushed anywhere yet. But I'm closing in on getting the patch ready.

Reply:
1. simd construct with memory loading has supported, dose the construct with gather need to be the extension?
2. In my mind, _Flags for __mem of gather is necessary

Reply:
if you have implemented _S_gather, maybe will I  based on your implementation be better?